packages/legend-application-studio/src/components/editor/side-bar/CreateNewElementModal.tsx

```diff
@@ -302,17 +274,50 @@ const NewConnectionDriverEditor = observer(() => {
     label: prettyCONSTName(currentConnectionType),
     value: currentConnectionType,
   };
-  const connectionOptions = Object.values(CONNECTION_TYPE).map((e) => ({
-    label: prettyCONSTName(e),
-    value: e,
-  }));
+  const extraOptionTypes = editorStore.pluginManager
+    .getApplicationPlugins()
+    .flatMap(
+      (plugin) =>
+        (
+          plugin as DSL_Mapping_LegendStudioApplicationPlugin_Extension
+        ).getExtraConnectionTypeOptions?.() ?? [],
+    );
+  const connectionOptions = [
+    PURE_MODEL_CONNECTION,
```

Comment:
Let's have the CONNECTION_TYPE enum and build everything from that.

Reply:
Thanks done